libs/std/stdio.d

```diff
@@ -2,8 +2,17 @@
 module std.stdio;
 
 
-void writeln(string s)
-{
-    printf(s);
-    printf("\n");
+	
+void writeln(string s) {
+	// NOTE sdc cannot take the pointer from string literals.
+	// Also, string literals are null terminated by default.
+	static immutable string formatString = "%.*s\n";
```

Comment:
Yes, dmd can do this. SDC compiles this code, and it cannot yet.
On Thu, Apr 2, 2015 at 11:56 AM deadalnix <notifications@github.com> wrote:

> In libs/std/stdio.d
> <https://github.com/deadalnix/SDC/pull/73#discussion_r27686474>:
>
> > @@ -2,8 +2,17 @@
> >  module std.stdio;
> >
> >
> > -void writeln(string s)
> > -{
> > -    printf(s);
> > -    printf("\n");
> > +	
> > +void writeln(string s) {
> > +	// NOTE sdc cannot take the pointer from string literals.
> > +	// Also, string literals are null terminated by default.
> > +	static immutable string formatString = "%.*s\n";
>
> That is false, and there is plenty of examples in SDC's codebase that use
> printf("foo".ptr);
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/deadalnix/SDC/pull/73/files#r27686474>.
>